include/deal.II/lac/constraint_matrix.templates.h

```diff
@@ -2269,8 +2274,13 @@ ConstraintMatrix::distribute_local_to_global (
   // an array in any case since we cannot know about the actual data type in
   // the ConstraintMatrix class (unless we do cast). This involves a little
   // bit of logic to determine the type of the matrix value.
-  std::vector<size_type> &cols = scratch_data->columns;
-  std::vector<number>     &vals = scratch_data->values;
+  std::vector<size_type> &cols          = scratch_data->columns;
+  std::vector<number>    &vals          = scratch_data->values;
+  // create arrays for writing into the vector as well
+  std::vector<size_type> &rows          = scratch_data->vector_indices;
```

Comment:
And on a completely different tangent, I don't find the use of aliases useful because it makes it so much more difficult to figure out who owns the object, what locking semantics are, etc. But that is for a separate patch.